### PR TITLE
Add minor comment for superfluid reward distr

### PR DESCRIPTION
**Pull request by mattverse**
## Description
Cref: https://github.com/osmosis-labs/osmosis/pull/1070#discussion_r824131325

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: x/superfluid/keeper/hooks_test.go [hunk 1]
```diff
@@ -16,6 +16,11 @@ func (suite *KeeperTestSuite) TestSuperfluidAfterEpochEnd() {
 			"happy path with single validator and delegator",
 			[]stakingtypes.BondStatus{stakingtypes.Bonded},
 			[]superfluidDelegation{{0, 0, 0, 1000000}},
+			// bond denom staked in pool = 15_000_000
+			// with risk adjustment, the actual bond denom staked via superfluid would be 15_000_000 * (1 - 0.5) = 10_000_000
```
Comment (ValarDragon) [inline x/superfluid/keeper/hooks_test.go:20]: isn't this 7_500_000. ?

**Comment by mattverse** [comment]
@ValarDragon  Fixed in addition to couple fixes!

File: x/superfluid/keeper/hooks_test.go (hunk 1 above)
Comment (mattverse) [reply x/superfluid/keeper/hooks_test.go]: Comments added for this!

File: app/apptesting/test_suite.go [hunk 2]
```diff
@@ -65,18 +65,18 @@ func (keeperTestHelper *KeeperTestHelper) SetupValidator(bondStatus stakingtypes
 	return valAddr
 }
 
-func (keeperTestHelper *KeeperTestHelper) BeginNewBlock(executeNextEpoch bool) {
+func (keeperTestHelper *KeeperTestHelper) BeginNewBlock(executeNextEpoch bool, proposer sdk.ValAddress) {
```
Comment (ValarDragon) [inline app/apptesting/test_suite.go:68]: Perhaps we should make a `BeginNewBlock` and a `BeginNewBlockWithProposer` functions? I imagine in a lot of cases, we don't want to deal with the caller choosing the proposer

File: x/superfluid/keeper/keeper_test.go [hunk 3]
```diff
@@ -36,7 +36,6 @@ func (suite *KeeperTestSuite) SetupTest() {
 	queryHelper := baseapp.NewQueryServerTestHelper(suite.Ctx, suite.App.InterfaceRegistry())
 	types.RegisterQueryServer(queryHelper, suite.App.SuperfluidKeeper)
 	suite.queryClient = types.NewQueryClient(queryHelper)
-	suite.SetupDefaultPool()
```
[the remark this answers is not in the document]
Comment (ValarDragon) [reply x/superfluid/keeper/keeper_test.go:39]: Agreed, manually creating pools is better

File: x/superfluid/keeper/hooks_test.go [hunk 4]
```diff
@@ -67,8 +89,8 @@ func (suite *KeeperTestSuite) TestSuperfluidAfterEpochEnd() {
 				suite.Require().True(found)
 				suite.Require().Equal(sdk.NewDec(7500000), delegation.Shares)
 			}
-			balance := suite.App.BankKeeper.GetAllBalances(suite.Ctx, delAddrs[0])
-			suite.Require().Equal(tc.expRewards, balance)
+			//balance := suite.App.BankKeeper.GetAllBalances(suite.Ctx, delAddrs[0])
+			//suite.Require().Equal(tc.expRewards, balance)
```
Comment (ValarDragon) [inline x/superfluid/keeper/hooks_test.go:93]: Is this intentional? Isn't this the check for expected rewards at the end?

File: app/apptesting/test_suite.go [hunk 5]
```diff
@@ -104,6 +104,43 @@ func (keeperTestHelper *KeeperTestHelper) BeginNewBlock(executeNextEpoch bool) {
...
 }
 
+func (keeperTestHelper *KeeperTestHelper) BeginNewBlockWithProposer(executeNextEpoch bool, proposer sdk.ValAddress) {
+	valAddr := []byte(":^) at this distribution workaround")
+	validator, found := keeperTestHelper.App.StakingKeeper.GetValidator(keeperTestHelper.Ctx, proposer)
+	if !found {
+		valAddrFancy := keeperTestHelper.SetupValidator(stakingtypes.Bonded)
+		validator, _ := keeperTestHelper.App.StakingKeeper.GetValidator(keeperTestHelper.Ctx, valAddrFancy)
+		valAddr2, _ := validator.GetConsAddr()
+		valAddr = valAddr2.Bytes()
```
Comment (ValarDragon) [inline app/apptesting/test_suite.go:114]: Should this return an error if the provided proposer doesn't exist instead?

Comment (ValarDragon) [reply app/apptesting/test_suite.go:114]: Then we can also have BeginNewBlock basically do
```
if len(validators) > 1 {
   BeginNewBlockWithProposer(executeNextEpochBlock, validators[0].Address)
}
// Create validator code
BeginNewBlockWithProposer(executeNextEpochBlock, newValidatorAddress)

File: x/superfluid/keeper/hooks_test.go (hunk 4 above)
Comment (alexanderbez) [reply x/superfluid/keeper/hooks_test.go:93]: We shouldn't have commented out code in general :P

**Review by ValarDragon — APPROVED** [review]
LGTM once linter passes.
We may need to do `var valAddr []byte`, and remove the ineffassign , or add a nolint

File: x/superfluid/keeper/hooks_test.go [hunk 6]
```diff
@@ -67,8 +89,12 @@ func (suite *KeeperTestSuite) TestSuperfluidAfterEpochEnd() {
 				suite.Require().True(found)
 				suite.Require().Equal(sdk.NewDec(7500000), delegation.Shares)
 			}
-			balance := suite.App.BankKeeper.GetAllBalances(suite.Ctx, delAddrs[0])
-			suite.Require().Equal(tc.expRewards, balance)
+
+			// check balances
```
Comment (alexanderbez) [inline x/superfluid/keeper/hooks_test.go:93]:
```suggestion
```

File: app/apptesting/test_suite.go [hunk 7]
```diff
@@ -78,6 +78,18 @@ func (keeperTestHelper *KeeperTestHelper) BeginNewBlock(executeNextEpoch bool) {
...
 		valAddr = valAddr2.Bytes()
 	}
+	keeperTestHelper.BeginNewBlockWithProposer(executeNextEpoch, valAddr)
+}
+
+func (keeperTestHelper *KeeperTestHelper) BeginNewBlockWithProposer(executeNextEpoch bool, proposer sdk.ValAddress) {
+	valAddr := []byte(":^) at this distribution workaround")
+	validator, found := keeperTestHelper.App.StakingKeeper.GetValidator(keeperTestHelper.Ctx, proposer)
+	if !found {
+		panic("given validator not found")
```
Comment (alexanderbez) [inline app/apptesting/test_suite.go:88]: Why panic instead of using a `*testing.T` assertion?

File: app/apptesting/test_suite.go [hunk 8]
```diff
@@ -78,6 +78,18 @@ func (keeperTestHelper *KeeperTestHelper) BeginNewBlock(executeNextEpoch bool) {
 		valAddr2, _ := validator.GetConsAddr()
 		valAddr = valAddr2.Bytes()
 	}
+	keeperTestHelper.BeginNewBlockWithProposer(executeNextEpoch, valAddr)
+}
+
+func (keeperTestHelper *KeeperTestHelper) BeginNewBlockWithProposer(executeNextEpoch bool, proposer sdk.ValAddress) {
+	valAddr := []byte(":^) at this distribution workaround")
```
Comment (alexanderbez) [inline app/apptesting/test_suite.go:85]: What's the context behind this, just curious?

**Review by alexanderbez — APPROVED** [review]

File: app/apptesting/test_suite.go (hunk 8 above)
Comment (ValarDragon) [reply app/apptesting/test_suite.go:85]: oh it was referring to it being hard to simply simulate an app.go begin block for tests (thats what all this function is doing) It was a super late night for superfluid staking, trying to get this test working lol, we should probably delete the message.

e.g. this + the last commit info was really confusing to get working (distribution would panic w/o this)
```
valAddr := []byte(":^) at this distribution workaround") //nolint
	validators := keeperTestHelper.App.StakingKeeper.GetAllValidators(keeperTestHelper.Ctx)
	if len(validators) >= 1 {
		valAddrFancy, err := validators[0].GetConsAddr()
		keeperTestHelper.Require().NoError(err)
		valAddr = valAddrFancy.Bytes()
	} else {
		valAddrFancy := keeperTestHelper.SetupValidator(stakingtypes.Bonded)
		validator, _ := keeperTestHelper.App.StakingKeeper.GetValidator(keeperTestHelper.Ctx, valAddrFancy)
		valAddr2, _ := validator.GetConsAddr()
		valAddr = valAddr2.Bytes()
	}valAddr := []byte(":^) at this distribution workaround") //nolint
	validators := keeperTestHelper.App.StakingKeeper.GetAllValidators(keeperTestHelper.Ctx)
	if len(validators) >= 1 {
		valAddrFancy, err := validators[0].GetConsAddr()
		keeperTestHelper.Require().NoError(err)
		valAddr = valAddrFancy.Bytes()
	} else {
		valAddrFancy := keeperTestHelper.SetupValidator(stakingtypes.Bonded)
		validator, _ := keeperTestHelper.App.StakingKeeper.GetValidator(keeperTestHelper.Ctx, valAddrFancy)
		valAddr2, _ := validator.GetConsAddr()
		valAddr = valAddr2.Bytes()
	}
```

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**